### ISSUE TITLE
get_string not accepting input

### ISSUE THREAD

**Issue by adams-x0** [Begintocodewithpython/samples#5]
When I run this code,  I can't type in my name  in the pygame window or idle shell, the star(*) in (Enter your name: *) just keeps blinking.
please help

**Comment by banksy0110** [comment]
same question ，have you solved it ?  could you share me how to solving it？

**Comment by adams-x0** [comment]
No, I haven't  solved itOn Dec 10, 2021 9:48 PM, banksy0110 ***@***.***> wrote:

same question ，have you solved it ?  could you share me how to solving it？

**Comment by coryac1** [comment]
Same question.  It seems that perhaps the functions in the snaps library are not compatible with the later versions of Python.  I also have some issues with display_text.

**Comment by coryac1** [comment]
Is there an updated version of the snaps library?  I am using the one from 2017, version 1.0.

**Comment by coryac1** [comment]
Here is the code for the get_string function in snaps.  When I just run this after importing snaps, it tells me setup() on line 12 is not defined.  Perhaps a clue for a more advanced programmer.

(for reference, here are the versions I am using:  pygame 2.6.1 (SDL 2.28.4, Python 3.13.7))

def get_string(prompt, size=50, margin=20,
               color=(255, 0, 0), horiz='left', vert='center',
               max_line_length=20):
    '''
    Reads a string from the user
    '''

    setup()

    result = ''
    cursor_char = '*'
    cursor = None
    
    def redraw():
        clear_display()
       
        render_message(prompt+result, margin=margin, size=size,
                       horiz=horiz, vert=vert, color=color, cursor=cursor)

    def cursor_flip():
        nonlocal cursor

    # create a timer for the cursor

    cursor_event = pygame.USEREVENT+1
    
    pygame.time.set_timer(cursor_event,500)

    while True:
        event = pygame.event.wait()

        if event.type == cursor_event:
            if cursor:
                cursor = None
            else:
                cursor = cursor_char
            redraw()
        elif event.type == 2:
            # Event 2 is keydown
            key_code = event.dict['unicode']
            if key_code is None:
                continue
            if key_code == '\r':
                break
            elif key_code == '\x08':
                if len(result) > 0:
                    result=result[:-1]
                    redraw()
            else:
                if len(result) < max_line_length:
                    result += key_code
                    redraw()

    # disable the timer for the cursor
    pygame.time.set_timer(cursor_event,0)
    return result

import urllib.request
import xml.etree.ElementTree

**Comment by coryac1** [comment]
Tried running the program directly in Python.  Keyboard remains unresponsive and cannot enter name.  Same with EG5-13 Theme Park Snaps Display, cannot enter ride number.
